docs/cudf/source/user_guide/benchmarks.ipynb

```diff
@@ -0,0 +1,1059 @@
+       "      <td>Buckees</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>4</th>\n",
+       "      <td>-165</td>\n",
+       "      <td>Walmart</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>...</th>\n",
+       "      <td>...</td>\n",
+       "      <td>...</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>9999995</th>\n",
+       "      <td>908</td>\n",
+       "      <td>Costco</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>9999996</th>\n",
+       "      <td>338</td>\n",
+       "      <td>McD</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>9999997</th>\n",
+       "      <td>885</td>\n",
+       "      <td>Buckees</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>9999998</th>\n",
+       "      <td>-391</td>\n",
+       "      <td>McD</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>9999999</th>\n",
+       "      <td>857</td>\n",
+       "      <td>Walmart</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "<p>10000000 rows × 2 columns</p>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "         numbers business\n",
+       "0           -316  Buckees\n",
+       "1           -441   Costco\n",
+       "2            653      McD\n",
+       "3            216  Buckees\n",
+       "4           -165  Walmart\n",
+       "...          ...      ...\n",
+       "9999995      908   Costco\n",
+       "9999996      338      McD\n",
+       "9999997      885  Buckees\n",
+       "9999998     -391      McD\n",
+       "9999999      857  Walmart\n",
+       "\n",
+       "[10000000 rows x 2 columns]"
+      ]
+     },
+     "execution_count": 6,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "gdf"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def time_it(callable):\n",
+    "    start = time.time()\n",
+    "    callable()\n",
+    "    end = time.time()\n",
+    "    return end - start"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "start = time.time()\n",
+    "pdf.to_parquet(\"pandas.parquet\")\n",
+    "end = time.time()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pandas_to_parquet = time_it(lambda : pdf.to_parquet(\"pandas.parquet\"))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "cudf_to_parquet = time_it(lambda : gdf.to_parquet(\"cudf.parquet\"))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pandas_read_parquet = time_it(lambda : pd.read_parquet(\"pandas.parquet\"))"
```

Comment:
For the reader functions, we need to consider cache clearing. I always use this command when benchmarking file systems: `os.system("/sbin/sysctl vm.drop_caches=3")`

Also, we don't know where the file is actually going for the user that runs this! It could be a local drive, it could be a network drive, it could be a virtual drive that is actually a network drive. Generally faster drives allow our readers to show faster speedups. We may want to compare host buffers and files, and provide some analysis based on the comparison.

All that said, for the purposes of this notebook I think your approach is the best one. We could include more documentation about how IO works. We could also add a drive read speed test, and report performance relative to that. I'm looking forward to discussing this more with you.